src/main/kotlin/com/sourcegraph/cody/chat/ui/MessagePart.kt

```diff
@@ -59,9 +61,21 @@ class CodeEditorPart(
     }
   }
 
-  private fun updateText(project: Project, text: String) {
-    this.text = text
-    WriteCommandAction.runWriteCommandAction(
-        project, Computable { editor.document.replaceText(text, System.currentTimeMillis()) })
+  private fun updateText(project: Project, newText: String) {
+    val newIndex = newText.length - 1
+    if (newIndex >= lastIndex) {
+      val size = newIndex - lastIndex
+      if (size > 0) {
+        val toAdd = newText.substring(lastIndex, newIndex)
+        WriteCommandAction.runWriteCommandAction(
+            project, Computable { editor.document.insertString(lastIndex, toAdd) })
+        lastIndex = newIndex
+      } else {
+        // todo sometimes difference to previous response may be 0 (responses have no value / don't
+        // need to refresh)
+      }
+    } else {
+      // todo this may happen (cody responses are not always ordered)
```

Comment:
that sounds a bit strange, let's discuss offline.

Reply:
Discussed. There is a different implementation now.